src/utils.js

```diff
@@ -18,6 +18,22 @@ genotet.utils.PIXEL_TOLERANCE = .1;
 /** @const {number} */
 genotet.utils.RANGE_TOLERANCE = .001;
 
+/** @const {RegExp} */
+genotet.utils.VALIDATE_EMAIL_REGEX =
+  /^([a-zA-Z0-9_\.\-])+\@(([a-zA-Z0-9\-])+\.)+([a-zA-Z0-9]{2,4})+$/;
+
+/** @const {RegExp} */
+genotet.utils.VALIDATE_USERNAME_REGEX = /^\w+$/;
```

Comment:
If usernames are not case sensitive, the you need to prevent the user from
entering uppercase letters.

On Sunday, March 20, 2016, Liana Yang <notifications@github.com> wrote:

> In src/utils.js
> <https://github.com/ViDA-NYU/genotet/pull/145#discussion_r56759459>:
>
> > @@ -18,6 +18,22 @@ genotet.utils.PIXEL_TOLERANCE = .1;
> >  /** @const {number} */
> >  genotet.utils.RANGE_TOLERANCE = .001;
> >
> > +/** @const {RegExp} */
> > +genotet.utils.VALIDATE_EMAIL_REGEX =
> > +  /^([a-zA-Z0-9_\.\-])+\@(([a-zA-Z0-9\-])+\.)+([a-zA-Z0-9]{2,4})+$/;
> > +
> > +/** @const {RegExp} */
> > +genotet.utils.VALIDATE_USERNAME_REGEX = /^\w+$/;
>
> They are not case sensitive. Why also lowercase them on the server? I
> don't think it is necessary.
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/ViDA-NYU/genotet/pull/145/files/83dd548aa3090df14c14b115859530f16cb2a2a5#r56759459>
>